api/apiclient.go

```diff
@@ -487,6 +498,24 @@ type dialOpts struct {
 	certPool *x509.CertPool
 }
 
+// parseURLWithOptionalScheme parses an input string
+// that may exclude a scheme, i.e. missing "http://".
+// If no scheme is provided, the default scheme
+// for the apiserver will be used.
+func parseURLWithOptionalScheme(addr string) (*url.URL, bool) {
+	// Add the schema if parsing fails or the host is empty.
+	// This avoids parsing ambiguity in url.Parse.
+	// We default to secure websockets (wss).
+	url, err := url.Parse(addr)
+	if err != nil || url.Host == "" {
```

Comment:
If err != nil here, adding a scheme to the front of the invalid addr ain't going to help?